tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -572,6 +572,32 @@ def _test_cpu_wrmsr_restore(context):
     microvm_factory = context.custom["microvm_factory"]
     cpu_template = context.custom["cpu_template"]
 
+    cpu_template_dir = cpu_template if cpu_template else "none"
+    snapshot_artifacts_dir = (
+        Path(shared_names["snapshot_artifacts_root_dir_wrmsr"])
+        / context.kernel.base_name()
+        / cpu_template_dir
+    )
+
+    # It is possible that some X template is not supported on
+    # the instance where the snapshots were created and,
+    # snapshot is loaded on an instance where X is supported. This
+    # results in error since restore doesn't find the file to load.
+    # for e.g. lets suppose snapshot is created on Skylake and restored
+    # on Cascade Lake. So, the created artifacts could just be:
+    # snapshot_artifacts/wrmsr/vmlinux-4.14/T2S
+    # but the restore test would fail because the files in
+    # snapshot_artifacts/wrmsr/vmlinux-4.14/T2CL won't be available.
+    # To avoid this we make an assumption that if template directory
+    # does not exist then snapshot was not created for that template
+    # and we skip the test.
+    if not Path.exists(snapshot_artifacts_dir):
+        reason = f"\n Since {snapshot_artifacts_dir} does not exist \
+                we skip the test assuming that snapshot was not     \
+                created for {cpu_template=}"
+        print(re.sub(" +", " ", reason))
```

Comment:
nit: I don't think I saw something like that for a multiline string before. Could be there a more idiomatic way of doing that?
Also, my impression is we don't tend to use `print()` in tests, because they may not be visible, unless pytest is run with `-s`. Is using  logger an option?

Reply:
logger is added but to use it we would have to modify all the buildkite jobs to have --log-cli-level=INFO
for the multiline string comment, re.sub seems to be the standard way to remove multiple spaces from a string.

Reply:
pytest seems to be the preferred way so going with that.